app/services/generate_coupon_service.rb

```diff
@@ -0,0 +1,11 @@
+class GenerateCouponService
+  class << self
+    def generate
+      loop do
+        coupon = Array.new(10) { rand(1...10) }.join
+
+        return coupon if Coupon.where(coupon: coupon).empty?
```

Comment:
[Readme](https://andreigridnev.com/blog/2016-04-17-empty-blank-any-exists-methods-of-ruby-on-rails-activerecord/)
use ```any?``` instead of  ```.empty?```